File: swift/Sources/PrintDate/PrintDate.swift
import Foundation

public struct PrintDate {
    private let calendar: Calendar;
    private let printer: Printer;

    public init(calendar: Calendar, printer: Printer) {
        self.calendar = calendar;
        self.printer = printer;
    }

    public func print_current_date() {
        let format = DateFormatter()
        format.timeStyle = .long
        format.dateStyle = .none
        let today = self.calendar.today()
        self.printer.print_line(line: format.string(from: today))
    }
}

File: java/src/main/java/team/codium/legacytraining/printdate/PrintDate.java
package team.codium.legacytraining.printdate;

public class PrintDate {

	private final Calendar calendar;
	private final Printer printer;

	public PrintDate(Calendar calendar, Printer printer) {
		this.calendar = calendar;
		this.printer = printer;
	}

	public void printCurrentDate() {
		String line = calendar.today().toString();
		printer.printLine(line);
	}

}
